src/editorutil.ts

```diff
@@ -23,4 +42,72 @@ export class EditorUtil {
             }
         });
     }
+
+    public async openServerJsonResponse(content: Protocol.GetServerJsonResponse): Promise<void> {
+        if (!content || !content.serverHandle || !content.serverJson) {
+            return;
+        }
+
+        if (this.serverTmpFiles.has(content.serverHandle.id)) {
+            this.saveTmpJsonresponse(
+                this.serverTmpFiles.get(content.serverHandle.id),
+                content
+            );
+        } else {
+            tmp.file({ prefix: 'tmpServerConnectorProp', postfix: '.json' }, (err, path, fd) => {
+                this.serverTmpFiles.set(content.serverHandle.id, path);
+                this.saveTmpJsonresponse(path, content);
+            });
+        }
+    }
+
+    private async saveTmpJsonresponse(path: string, content: Protocol.GetServerJsonResponse): Promise<void> {
+        fs.writeFile(path, content.serverJson, undefined, () => {
+            return Promise.reject();
+        });
+
+        vscode.workspace.openTextDocument(path).then(doc =>
+            vscode.window.showTextDocument(doc)
+        );
+    }
+
+    public async onDidSaveTextDocument(doc: vscode.TextDocument) {
+        if (!doc) {
+            return;
+        }
+        if (await this.checkTmpServerPropsFile(doc)) {
+            const serverId = await Utils.getKeyByValue<string>(this.serverTmpFiles, doc.uri.path);
+            if (!serverId) {
+                vscode.window.showErrorMessage('Unable to save server properties');
+                return;
+            }
+            const serverHandle: Protocol.ServerHandle = this.explorer.serverStatus.get(serverId).server;
+            if (!serverHandle) {
+                vscode.window.showErrorMessage('Unable to save server properties');
+                return;
+            }
+            this.explorer.saveServerProperties(serverHandle, doc.getText()).catch(error => {
+                vscode.window.showErrorMessage(`Unable to save server properties. ${error}`);
+            });
+        }
+    }
+
+    public async onDidCloseTextDocument(doc: vscode.TextDocument) {
+        if (!doc) {
+            return;
+        }
+        if (await this.checkTmpServerPropsFile(doc)) {
+            fs.unlink(doc.uri.fsPath, error => {
+                console.log(error);
+            });
+        }
+    }
+
+    private async checkTmpServerPropsFile(doc: vscode.TextDocument): Promise<boolean> {
```

Comment:
Why use a promise here? #indexOf does not return a promise, why wrap it in such?